report/report.py

```diff
@@ -126,10 +127,14 @@ def generate_report(args, *images):
     write_report(report)
 
 
-
 def execute_dockerfile(args):
     '''Execution path if given a dockerfile'''
     logger.debug('Setting up...')
+    try:
+        subprocess.check_output('docker ps', shell=True)
+    except subprocess.CalledProcessError as error:
+        logger.error('Docker daemon is not running: {0}'.format(error.output))
```

Comment:
Is error.output a byte stream or a string?

Reply:
It is a byte stream and I got my error :)

Reply:
I'm curious then - did you get the 'b' at the beginning of the error string? If yes then you will need to convert the byte stream into a string using error.output.decode('utf-8').

Reply:
Yes, I got 'b' and I will update it. Thanks

Reply:
No worries! Normally you would get an error saying you cannot convert a bytestring into a string but format does the !r conversion for you. It's nice to know exactly what types you're working with ^_^

Reply:
Yes, docker_command runs a shell command and raises a CalledProcessError is the command fails so you can put that in a try-except as you have done in report.py
>>> import subprocess
>>> from utils.container import docker_command
>>> try:
...     docker_command(['docker', 'ps'])
... except subprocess.CalledProcessError as error:
...     print(error.output.decode('utf-8'))
...
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?